### PR TITLE
💥 FIX:  Web server should be listening on port 80 💥

### PR DESCRIPTION
**Pull request by patricknelson**
Fixes the fact that [chaosthebot.com](http://chaosthebot.com) isn't working consistently despite the server already being online (see [chaosthebot.com:8080](http://chaosthebot.com:8080)).

Edit: It varies, it seems to be crashing/restarting repeatedly. See comment below from @amoffat https://github.com/chaosbot/chaos/pull/107#issuecomment-303548893

### CODE REVIEW

**Review by pchauncey — APPROVED** [review]

**Comment by sugar700** [comment]
There are actually two servers, not sure why the one on port 80 is down :-1:.

**Comment by sugar700** [comment]
This is the server I'm talking about - https://github.com/chaosbot/chaos/blob/master/server/server.py

**Comment by amoffat** [comment]
@xfix here's what I see

Starting once:
```
INFO:root:starting up and entering event loop
INFO:chaosbot:starting http server
INFO:chaosbot:looking for PRs
DEBUG:github_api:requests remaining: inf, reset in: 0s, cooldown sleep: 0.00s
INFO:github_api:requesting GET to '/repos/chaosbot/Chaos/pulls'
INFO:root:sleeping for 30 seconds
```

Restarting

```
INFO:root:starting up and entering event loop
INFO:chaosbot:starting http server
INFO:chaosbot:looking for PRs
DEBUG:github_api:requests remaining: inf, reset in: 0s, cooldown sleep: 0.00s
INFO:github_api:requesting GET to '/repos/chaosbot/Chaos/pulls'
Traceback (most recent call last):
  File "server.py", line 16, in <module>
    httpd = socketserver.TCPServer(("", PORT), Handler)
  File "/usr/lib/python3.6/socketserver.py", line 453, in __init__
    self.server_bind()
  File "/usr/lib/python3.6/socketserver.py", line 467, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 98] Address already in use
INFO:root:sleeping for 30 seconds
```

I believe the socket is in a TIME-WAIT state when chaos starts up and [kills it](https://github.com/chaosbot/chaos/blob/master/chaos.py#L62), and it can't listen again right away.  This is probably why it's alternating between listening and not listening as the server restarts over time.

**Comment by patricknelson** [comment]
@amoffat @xfix  maybe all the code responsible for starting the current server listening on 8080 should be removed/abstracted away to server.py or vice versa. Personally I'd suggest moving it into `chaos.py` since it's not much code anyway and easier to read (right now).
 
Disclaimer: I am not a python dev.

**Comment by patricknelson** [comment]
Also: Is the `pkill` statement really necessary if the script has already been terminated/restarted after a merge?

**Comment by patricknelson** [comment]
Flagging as `WIP` since I wanna DL and install a Vagrant image of Ubuntu 14.04 so I can test this out first hand (may result in another PR so other people can easily tinker with this in their own local dev too but... maybe not)

**Comment by tbodt** [comment]
👍

**Comment by patricknelson** [comment]
Ok, fixed everything, it's serving on port 80. Setup a vagrant VM to test and all is well.

Just gotta get this shit merged in ASAP before the core codebase changes too much again...

**Comment by patricknelson** [comment]
@amoffat do you have a local dev version of this you can test this on to ensure it works? I've tested this myself but in order to get it working I had to comment out all the github-related stuff 😢

**Comment by patricknelson** [comment]
@xfix @davidak @kmaschta I've adapted the `server.py` code, resolved conflicts with `master` branch and rolled that into this PR. Since I pushed new commits everyone has to re-vote again for it to get merged 😕 It should work now, try it out yourself in #123 (check that branch out, merge this one into that one and edit out the github api stuff unless you have your own secret to drop in) and you'll see 😄

File: chaos.py [hunk 1]
```diff
@@ -59,17 +62,14 @@ def main():
 
     log = logging.getLogger("chaosbot")
 
-    api = gh.API(settings.GITHUB_USER, settings.GITHUB_SECRET)
```
Comment (patricknelson) [inline chaos.py:62]: FYI, removed this line as it was no longer in use.

**Comment by ghost** [comment]
Since http://chaosthebot.com is working just fine on the standard port, this should probably be closed.

**Comment by g19fanatic** [comment]
#163 is related to this.
Setting up a forwarding proxy would alleviate all concerns on what port we're publishing content out of. 

Nginx would be port 80. All http(s) requests would be parsed and forward to the proper server behind it.

**Comment by patricknelson** [comment]
Ah, that's where you're wrong 😄 Note that it's currently down, likely due to the issue @amoffat first noted here: https://github.com/chaosbot/chaos/pull/107#issuecomment-303548893 which I've previously edited into my first comment above.

Nginx would be a good solution and since Puppet has been implemented thanks to @g19fanatic #115 I think that's a *much* better route (easily configurable).

**EDIT:** Closing this PR. Go vote for nginx! **https://github.com/chaosbot/chaos/pull/163**

**Pull request closed without merging**